packages/devtools_app/test/vm_developer/object_inspector/object_viewport_test.dart

```diff
@@ -144,16 +50,98 @@ void main() {
+        wrap(ObjectViewport(controller: testObjectInspectorViewController)),
+      );
+      expect(viewportTitle(mockClassObject), 'Class FooClass');
+      expect(find.text('Class FooClass'), findsOneWidget);
+      expect(find.byType(VmClassDisplay), findsOneWidget);
+    });
+  });
+
+  group('test for field object:', () {
+    late MockFieldObject mockFieldObject;
+
+    setUp(() {
+      mockFieldObject = MockFieldObject();
+
+      mockVmObject(mockFieldObject);
+      when(mockFieldObject.name).thenReturn(testField.name);
+      when(mockFieldObject.ref).thenReturn(testField);
+      when(mockFieldObject.obj).thenReturn(testField);
+      when(mockFieldObject.guardClass).thenReturn(null);
+      when(mockFieldObject.guardNullable).thenReturn(null);
+      when(mockFieldObject.guardClassKind).thenReturn(null);
+    });
+
+    testWidgets('viewport shows field display', (WidgetTester tester) async {
+      testObjectInspectorViewController.fakeObjectHistory
+          .setCurrentObject(mockFieldObject);
+
+      await tester.pumpWidget(
+        wrap(ObjectViewport(controller: testObjectInspectorViewController)),
+      );
+
+      expect(viewportTitle(mockFieldObject), 'Field fooField');
+      expect(find.text('Field fooField'), findsOneWidget);
+      expect(find.byType(VmFieldDisplay), findsOneWidget);
+    });
+  });
+
+  group('test for function object:', () {
+    late MockFuncObject mockFuncObject;
+
+    late Func testFunctionCopy;
+
+    setUp(() {
+      mockFuncObject = MockFuncObject();
+
+      final funcJson = testFunction.toJson();
+      testFunctionCopy = Func.parse(funcJson)!;
+
+      mockVmObject(mockFuncObject);
+      when(mockFuncObject.name).thenReturn(testFunctionCopy.name);
+      when(mockFuncObject.ref).thenReturn(testFunctionCopy);
+      when(mockFuncObject.obj).thenReturn(testFunctionCopy);
+      when(mockFuncObject.kind).thenReturn(null);
+      when(mockFuncObject.deoptimizations).thenReturn(null);
+      when(mockFuncObject.isOptimizable).thenReturn(null);
+      when(mockFuncObject.isInlinable).thenReturn(null);
+      when(mockFuncObject.hasIntrinsic).thenReturn(null);
+      when(mockFuncObject.isRecognized).thenReturn(null);
+      when(mockFuncObject.isNative).thenReturn(null);
+      when(mockFuncObject.vmName).thenReturn(null);
+      when(mockFuncObject.icDataArray).thenReturn(null);
+    });
+
+    testWidgets('viewport shows function display', (WidgetTester tester) async {
+      testObjectInspectorViewController.fakeObjectHistory
+          .setCurrentObject(mockFuncObject);
+
+      await tester.pumpWidget(
+        wrap(ObjectViewport(controller: testObjectInspectorViewController)),
+      );
+
+      expect(viewportTitle(mockFuncObject), 'Function fooFunction');
+      expect(find.text('Function fooFunction'), findsOneWidget);
+      expect(find.byType(VmFuncDisplay), findsOneWidget);
+    });
   });
 
   testWidgets('test for scriptObject', (WidgetTester tester) async {
```

Comment:
we will likely want each of these (script, library, instance) to be in their own group too

Reply:
Yeah, but I thought I would do that on their respective PRs. Did you want me to make the groups on this one?

Reply:
sg to do on their respective prs

Reply:
I made the group only for instance since that object type has not been implemented, the rest I'll do on their own PRs